### PR TITLE
Tests and fix for issue #254 (RunContainer.contains(ArrayContainer))

### PR DESCRIPTION
**Pull request by sesuncedu**
Legal-Meta - I didn't notice I didn't have sign-off set when I committed the change  to the test case, but by signing off on the substantive fix, and by submitting this pull request, it is my express intent to grant any and all rights necessary to incorporate these changes into the existing codebase, and that I am legally entitled to grant such rights under the laws of the United States of America and the State of North Carolina.
 
/s/ Simon Spero

### CODE REVIEW

File: roaringbitmap/src/main/java/org/roaringbitmap/RunContainer.java [hunk 1]
```diff
@@ -808,7 +808,7 @@ protected boolean contains(ArrayContainer arrayContainer) {
       return false;
     }
     int ia = 0, ir = 0;
-    while(ia < arrayContainer.getCardinality() && ir <= runCount) {
+    while(ia < arrayContainer.getCardinality() && ir < runCount) {
       int start = getValue(ir);
```
Comment (richardstartin) [inline roaringbitmap/src/main/java/org/roaringbitmap/RunContainer.java]: There is actually yet another error here: this should be 

```java 
int start = Util.toIntUnsigned(getValue(ir));
```

**Comment by richardstartin** [comment]
Good catch. There is another error in the method you are patching, perhaps you could add that fix in too while you are at it?

File: roaringbitmap/src/main/java/org/roaringbitmap/RunContainer.java [hunk 2]
```diff
@@ -808,7 +808,7 @@ protected boolean contains(ArrayContainer arrayContainer) {
...
       int stop = start + toIntUnsigned(getLength(ir));
       if(arrayContainer.content[ia] < start) {
```
Comment (richardstartin) [inline roaringbitmap/src/main/java/org/roaringbitmap/RunContainer.java]: There's another one here; we should be storing `arrayContainer.content[ia]` in an int variable:

```java
int value = Util.toIntUnsigned(arrayContainer.content[ia]);
if (value < start) {
...
} else if (value > stop) {
...```

**Comment by richardstartin** [comment]
LGTM

**Comment by lemire** [comment]
Taking into account  @richardstartin 's review... merging...

**Comment by lemire** [comment]
Updated https://github.com/RoaringBitmap/RoaringBitmap/blob/master/AUTHORS

**Pull request merged**